packages/dashboard/src/pages/Personas/components/ContactTableRow/index.tsx

```diff
@@ -0,0 +1,143 @@
+import type { Contact } from '@masknet/shared'
+import { memo, useCallback } from 'react'
+import { Box, TableCell, TableRow, Typography, makeStyles, Avatar, useTheme, Button } from '@material-ui/core'
+import { StarIcon, MaskNetworkIcon } from '@masknet/icons'
+import { MaskColorVar } from '@masknet/theme'
+import { Services } from '../../../../API'
+import { useDashboardI18N } from '../../../../locales'
+import { generateContactAvatarColor } from '../../../../utils/generateContactAvatarColor'
+import { useAddContactToFavorite, useRemoveContactFromFavorite } from '../../hooks/useFavoriteContact'
+import { PersonaContext } from '../../hooks/usePersonaContext'
+
+const useStyles = makeStyles(() => ({
+    favorite: {
+        marginLeft: 16,
+        marginRight: 26,
+        width: 24,
+        height: 24,
+    },
+    avatarContainer: {
+        position: 'relative',
+    },
+    maskIcon: {
+        position: 'absolute',
+        width: 16,
+        height: 16,
+        right: 0,
+        bottom: 0,
+    },
+    button: {
+        padding: '3px 24px',
+        borderRadius: 14,
+    },
+    info: {
+        display: 'flex',
+        flexDirection: 'column',
+        marginLeft: 17,
+        justifyContent: 'space-between',
+    },
+    userId: {
+        fontSize: 12,
+        color: MaskColorVar.normalText,
+    },
+}))
+
+export interface ContactTableRowProps {
+    contact: Contact
+    index: number
+    network: string
+}
+
+export const ContactTableRow = memo<ContactTableRowProps>(({ network, contact, index }) => {
```

Comment:
It does not `memo` if `contact` is an object and its identity changed after each request.

Reply:
This only changes the status of one `contact` when a favorite operation is performed and does not cause the status of all `contacts` to change.